test/unknown_state_handling_test.py

```diff
@@ -0,0 +1,83 @@
+from helpers import LuigiTestCase
+
+
+import luigi
+import luigi.worker
+import luigi.execution_summary
+
+
+class DummyRequires(luigi.Task):
+    def run(self):
+        print('just a dummy task')
+
+
+class FailInRun(luigi.Task):
+    def run(self):
+        print('failing in run')
+        raise Exception
+
+
+class FailInRequires(luigi.Task):
+    def requires(self):
+        print('failing')
+        raise Exception
+        return [DummyRequires()]
+
+    def run(self):
+        print('running')
+
+
+class FailInDepRequires(luigi.Task):
+    def requires(self):
+        return [FailInRequires()]
+
+    def run(self):
+        print('doing a thing')
+
+
+class FailInDepRun(luigi.Task):
+    def requires(self):
+        return [FailInRun()]
+
+    def run(self):
+        print('doing a thing')
+
+
+class UnknownStateTest(LuigiTestCase):
+    def setUp(self):
+        super(UnknownStateTest, self).setUp()
+        self.scheduler = luigi.scheduler.Scheduler(prune_on_get_work=False)
+        self.worker = luigi.worker.Worker(scheduler=self.scheduler)
```

Comment:
worker needs to be run with keep_alive=True for this test to actually exercise that the worker exits.